### ISSUE TITLE
some oneOf definitions fail to compile 

### ISSUE THREAD

**Issue by djm1329** [guardrail-dev/guardrail#1899]
Thanks  for all the recent effort towards a 1.0.0 release! 

This report concerns the new 1.0.0-SNAPSHOT candidate. 

Prior to 1.0.0-SNAPSHOT, the following oneOf used to get ignored, no model was generated, and guardrail compilation would succeed. In my use case this was ok since the particular component  was not used in my application. With 1.0.0-SNAPSHOT however I get an error when trying to run guardrail.  

Note the OpenAPI spec this is taken from is not under my control, it comes from a telco standards organization.

```    
  MnS:
      oneOf:
        - type: object
          properties:
            SubNetwork:
              $ref: '#/components/schemas/SubNetwork-Multiple'
        - type: object
          properties:
            ManagedElement:
              $ref: '#/components/schemas/ManagedElement-Multiple' 
``` 

```
Error:Only $ref is supported in oneOf at this time (.components.schemas.MnS)
```

Supporting objects as well as $ref in oneOf will be very valuable. In the meantime I wonder if it is possible to emit a warning instead of an error, and emit code that "falls back" to JSON, allowing guardrail code generation to succeed?

Thanks!

**Comment by djm1329** [comment]
@blast-hardcheese thank you, thank you, thank you!

**Comment by blast-hardcheese** [comment]
@djm1329 Thank you for the detailed report! This ended up paving the way for object literals elsewhere in specification files as well, so hopefully #76 will actually be able to be closed out now as well.

I've published new 1.0.0-SNAPSHOTs of core, scala-support, and java-support, they should be picked up.

Addendum, I just caught a bug when verifying in one of the sample projects, PR shortly.

**Comment by blast-hardcheese** [comment]
OK. Please pardon the back-to-back SNAPSHOT releases, #1903 for details.

As of now, there's one question as far as utility goes. Given the following specification chunk (given a valid `Pet` elsewhere), 

```yaml
    MnS:
      oneOf:
        - type: object
          required: [SubNetwork]
          properties:
            SubNetwork:
              $ref: '#/components/schemas/Pet'
        - type: object
          required: [ManagedElement]
          properties:
            ManagedElement:
              $ref: '#/components/schemas/Pet'
```

I can round-trip both `Nested1` and `Nested2` as follows:

```scala
scala> MnS(MnS.Nested1(Pet("Fluffy"))).asJson.as[MnS]
res6: io.circe.Decoder.Result[foo.definitions.MnS] = Right(Nested1(Nested1(Pet(Fluffy,None))))

scala> MnS(MnS.Nested1(Pet("Fluffy"))).asJson
res7: io.circe.Json =
{
  "SubNetwork" : {
    "name" : "Fluffy",
    "status" : null
  }
}

scala> MnS(MnS.Nested2(Pet("Fluffy"))).asJson.as[MnS]
res8: io.circe.Decoder.Result[foo.definitions.MnS] = Right(Nested2(Nested2(Pet(Fluffy,None))))

scala> MnS(MnS.Nested2(Pet("Fluffy"))).asJson
res9: io.circe.Json =
{
  "ManagedElement" : {
    "name" : "Fluffy",
    "status" : null
  }
}
```

_although,_ there's a subtle gotcha here. That being, if you both leave _off_ the `required` as well as have no `discriminator` defined, you get incoherence:

```scala
scala> MnS(MnS.Nested2(Some(Pet("Fluffy")))).asJson.as[MnS]
res1: io.circe.Decoder.Result[foo.definitions.MnS] = Right(Nested1(Nested1(None)))
```

Even though I encoded `Nested2` and decoded immediately, the decode did not produce the expected results due to `Nested1` being both the first decoder tried as well as comprised entirely of nullable fields, which of course is happy to ignore unknown properties and return an empty object.

@djm1329, Given the kinds of specifications you see, is this going to be a problem? There are alternate schemes that can be used here, but all of which are more complicated than what exists as the resolution to this issue.
